Lavinia-api/Utilities/ModelBuilder.cs

```diff
@@ -126,5 +138,91 @@ public static List<Result> BuildResults(IEnumerable<ResultFormat> results)
 
             return resultModels;
         }
+
+        // API v2
+        /// <summary>
+        /// Takes a list of CountyDataFormat and builds a list of DistrictMetrics
+        /// </summary>
+        /// <param name="countyData">A list of CountyDataFormat</param>
+        /// <returns>A list of DistrictMetrics</returns>
+        public static IEnumerable<DistrictMetrics> BuildDistrictMetrics(IEnumerable<CountyDataFormat> countyData)
+        {
+            return countyData.Select(data => new DistrictMetrics
+                {Area = data.Area, District = data.County, ElectionYear = data.Year, Population = data.Population, Seats = data.Seats});
+        }
+
+        public static IEnumerable<ElectionParameters> BuildElectionParameters(IEnumerable<ElectionFormat> electionData,
+            string electionType, IEnumerable<DistrictMetrics> districtMetrics)
+        {
+            return electionData.Select(data =>
+            {
+                List<ListElement<int>> districtSeats = new List<ListElement<int>>{new ListElement<int>
+                {
+                    Key = "SUM",
+                    Value = data.Seats
+                }};
+
+                if (data.AreaFactor < 0)
+                {
+                    districtSeats.AddRange(
+                        districtMetrics.Where(dM => dM.ElectionYear == data.Year)
+                            .Select(dM => new ListElement<int>
+                                {
+                                    Key = dM.District,
+                                    Value = dM.Seats
+                                }));
+                }
+
+
+                return new ElectionParameters
+                {
+                    Algorithm = BuildAlgorithmParameters(data),
+                    AreaFactor = data.AreaFactor,
+                    DistrictSeats = districtSeats,
+                    ElectionType = electionType,
+                    ElectionYear = data.Year,
+                    LevelingSeats = data.LevelingSeats,
+                    Threshold = data.Threshold,
+                    TotalVotes = 0
+                };
+            });
+        }
+
+        public static AlgorithmParameters BuildAlgorithmParameters(ElectionFormat data)
```

Comment:
Missing comments, this is a recurring thing that should be fixed before PR approval.